android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -100,4 +115,26 @@ internal data class SuggestionDetails(
     val fullKeyword: String,
     val isSponsored: Boolean,
     val icon: List<UByte>?,
+    val clickInfo: FxSuggestClickInfo? = null,
 )
+
+/**
+ * Collective of fields required for fxsuggest click telemetry
+ */
+sealed interface FxSuggestClickInfo {
+    /**
+     * AMP related fields for fx suggest click telemetry
+     *
+     * @param blockId A unique identifier for the suggestion.
+     * @param advertiser The name of the advertiser providing the sponsored suggestion.
+     * @param clickUrl The url to report the click to.
+     * @param contextId The categorization of the suggestion.
```

Comment:
```suggestion
     * @param iabCategory The categorization of the suggestion.
     * @param contextId The contextual services user identifier.
```

Sometimes the linter complains about out-of-date `@param` declarations, but it's green on yours!  🤷🏼‍♀️